misago/conf/admin/forms/oauth2.py

```diff
@@ -203,6 +206,26 @@ class OAuth2SettingsForm(SettingsForm):
         max_length=200,
         required=False,
     )
+    oauth2_enable_pkce = YesNoSwitch(
+        label=pgettext_lazy("admin oauth2 settings form", "Enable OAuth2 PKCE"),
+        help_text=pgettext_lazy(
+            "admin oauth2 settings form",
+            "Enabling OAuth2 PKCE (Proof Key for Code Exchange) will add additional protection during code to token exchanges if OAuth2 server supports it.",
+        ),
+        required=False,
+    )
+    oauth2_pkce_code_challenge_method = forms.ChoiceField(
+        label=pgettext_lazy("admin oauth2 settings form", "PKCE Code Challenge method"),
+        help_text=pgettext_lazy(
+            "admin oauth2 settings form",
+            "When PKCE is enabled this hashing method is used to generate the PKCE code challenge for OAuth2.",
+        ),
+        choices=[
+            ("S256", "S256"),
+            ("plain", "plain (no hashing)"),
```

Comment:
"plain" choice should be marked for translation. You can use `admin oauth2 settings pkce choice` as a context.